adapter/internal/adapter/adapter.go

```diff
@@ -277,16 +287,19 @@ func fetchAPIsOnStartUp(conf *config.Config) {
 		if data.Resp != nil {
 			// For successfull fetches, data.Resp would return a byte slice with API project(s)
 			logger.LoggerMgw.Debug("Pushing data to router and enforcer")
+			health.SetControlPlaneRestAPIStatus(true)
 			err := synchronizer.PushAPIProjects(data.Resp, envs)
 			if err != nil {
 				logger.LoggerMgw.Errorf("Error occurred while pushing API data: %v ", err)
 			}
 		} else if data.ErrorCode >= 400 && data.ErrorCode < 500 {
 			logger.LoggerMgw.Errorf("Error occurred when retrieveing APIs from control plane: %v", data.Err)
+			health.SetControlPlaneRestAPIStatus(true)
```

Comment:
In case of client side error, still the APIs we achieve is zero right. Unless APIM says there are zero APIS in the response message, we should fail health of adapter I guess, @pubudu538 wdyt?

Reply:
If there are no APIs for the gateway environment, APIM responds with 404, in that case, we should not set unhealthy. We chan check for `data.ErrorCode == 404` for that, @Rajith90 wdyt?

Reply:
for that case API response body explicitly says that no APIs found, we can check for that message and starts the adapter. And for all the other cases we can make the health false for 4xx error codes. WDYT?

Reply:
+1